backend/src/controllers/encounter.controller.ts

```diff
@@ -2,40 +2,79 @@
  */
  import { NextFunction, Request, Response } from 'express';
-
- import EncounterModel from '../models/encounter.model';
  import encounterService from '../services/encounter.service';
  import logger from '../utils/logger';
+import httpStatus from "http-status";
+import {EncounterModel} from "../models/encounter.model";
+import {getUserByAuthId} from "../services/user.service";
  
  export const createEncounter = async (
    req: Request,
    res: Response,
    next: NextFunction,
  ): Promise<void> => {
-   logger.info("POST /encounter/create request from frontend");
+   logger.info("POST /encounter request from frontend");
  
    try {
      // Grab the data from the req
      const encounterReq = getEncounterFromReqBody(req.body);
- 
+
      // Pass data to service and attempt to save
      const createdEncounter = await encounterService.createEncounter(encounterReq);
- 
      // Notify frontend that the operation was successful
-     res.sendStatus(200);
+     res.status(httpStatus.CREATED).json(createdEncounter).end();
    } catch (e) {
      next(e);
    }
  };
+export const updateEncounter = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+): Promise<any> => {
+    logger.info("PUT /encounter/:id request from frontend");
+
+    try {
+        // check authorization is not a null object (this is necessary check for lint)
+        if (req.headers.authorization == null)
+            return res.status(httpStatus.NOT_FOUND).end()
+
+        const encounterIdToUpdate = req.params.id
+        const firebaseAuthId = req.headers.authorization['user_id']
+
+        // check that user exists with that firebase auth_id, and the user specified encounter actually exists
+        const userByAuthId = await getUserByAuthId(firebaseAuthId);
+        if(!userByAuthId || !isEncounterExistsInUser(userByAuthId, encounterIdToUpdate))
+            return res.status(httpStatus.NOT_FOUND).end()
+
+        // update encounter
+        const newEncounterData = getEncounterFromReqBody(req.body);
+        const updatedEncounter = await encounterService.updateEncounter(encounterIdToUpdate, newEncounterData);
+        return res.sendStatus(updatedEncounter?httpStatus.NO_CONTENT:httpStatus.NOT_FOUND).end();
+    } catch (e) {
+        next(e);
+    }
+};
+
  
  // Util function that won't be needed regularly
 const getEncounterFromReqBody = (body: any) => {
-    const encounter = {
+    const encounter: EncounterModel = {
         date: body.date,
         location: body.location,
         description: body.description,
         persons: body.persons
     }
 
    return encounter;
- }
\ No newline at end of file
+ }
+
+/**
+ * searches for encounter in user by encounter id
+ * @param user user which encounter should be found from
+ * @param encounterId id of the encounter to be searched from the user
+ */
+const isEncounterExistsInUser = (user, encounterId: string): boolean => {
```

Comment:
Do you think we could improve this method name? Perhaps `doesEncounterExistInUser()` or `userContainsEncounter()`? wdyt?

Reply:
Thanks for your input!

The [convention](https://stackoverflow.com/a/5326487) I'm using is that a function that returns boolean is prefixed with `is`. Having `is` prefix lets devs know that it's a boolean function, which is useful when you construct branching logic since you know its return values are either true or false. I'm a fan of autocompletion, and it's _exponentially_ easier to pick up on a particular boolean functions if you know that it starts with `is`. I will probably keep this unless there's a good reason not to.

Reply:
Thanks for your explanation, I'm happy to leave the method as is! Do you think we should try using that convention project-wide?

Reply:
Maybe for new codes, but personally I don't see much use to creating a convention for just that - it's pretty rare to have a pure boolean function in TS. We can definitely consider it if we decide to come up with a selection of naming conventions!